frontend/src/Components/modals/index.ts

```diff
@@ -0,0 +1,295 @@
+
+// User Exports
+export const checkOnlyLettersAndSpaces: Validate<string, string | boolean> = (
+    input: string
+) => {
+    if (!/^[A-Za-z\s]+$/.test(input)) {
+        return 'Input should only contain letters and spaces';
+    }
+    return true;
+};
+export const checkOnlyLettersAndNumbers: Validate<string, string | boolean> = (
+    input: string
+) => {
+    if (!/^[A-Za-z0-9]+$/.test(input)) {
+        return 'Input should only contain letters and numbers';
+    }
+    return true;
+};
+export const userInputs: InputWithOptions<ProviderPlatform>[] = [
+    {
+        type: FormInputTypes.Text,
+        label: 'First Name',
+        interfaceRef: 'name_first',
+        required: true,
+        length: 25,
+        validate: checkOnlyLettersAndSpaces
+    },
+    {
+        type: FormInputTypes.Text,
+        label: 'Last Name',
+        interfaceRef: 'name_last',
+        required: true,
+        length: 25,
+        validate: checkOnlyLettersAndSpaces
+    },
+    {
+        type: FormInputTypes.Text,
+        label: 'Username',
+        interfaceRef: 'username',
+        required: true,
+        length: 50,
+        pattern: {
+            value: /^[A-Za-z0-9]+$/,
+            message:
+                'Username can only contain letters and numbers without spaces'
+        }
+    },
+    {
+        type: FormInputTypes.Text,
+        label: 'Email (optional)',
+        interfaceRef: 'email',
+        required: false,
+        length: 50
+    }
+];
+export { AddUserModal } from './AddUserModal';
+export { EditUserModal } from './EditUserModal';
+
+// Video Exports
+
+const getInvalidValidURLs = (urls: string[]): string[] | undefined => {
+    const badLinks: string[] = [];
+    urls.forEach((url) => {
+        if (
+            url.includes('youtube') &&
+            !url.includes('watch?v=') &&
+            !url.includes('youtu.be')
+        ) {
+            badLinks.push(url);
+        }
+    });
+    return badLinks.length > 0 ? badLinks : undefined;
+};
+
+export const checkValidVideoURLs: Validate<string, string> = (
+    input: string
+) => {
+    const urls = input
+        .split(',')
+        .map((url) => url.trim())
+        .filter((url) => url !== '');
+    if (urls.length === 0) {
+        return 'Please enter at least one valid URL.';
+    }
+    const badLinks = getInvalidValidURLs(urls);
+    if (badLinks != undefined) {
+        const badLinksString = `The following links entered are not valid Youtube links: ${badLinks.join(', ')}`;
+        return badLinksString;
+    }
+    return true;
+};
+
+export const videoInputs: Input[] = [
+    {
+        type: FormInputTypes.TextArea,
+        label: `This will begin to attempt to download videos. You can enter
+        links from youtube or any other popular video hosting site.
+        If there is an error, downloads will be attempted up to 5
+        times, after which time you will need to re-add the video.`,
```

Comment:
Can you shorten this message to

"Enter video URLs from YouTube and similar sites to download. If a download fails, it will be retried up to 5 times before the video needs to be added again."